ols/app/models/config.py

```diff
@@ -36,6 +36,32 @@ def _get_attribute_from_file(data: dict, file_name_key: str) -> Optional[str]:
     return None
 
 
+def _read_secret(
+    data: dict, path_key: str, default_filename: str, raise_on_error: bool = True
+) -> Optional[str]:
+    """Read secret from file on given path or from filename if path points to directory."""
+    path = data.get(path_key)
+
+    if path is None:
+        return None
+
+    filename = path
+    if os.path.isdir(path):
+        filename = os.path.join(path, default_filename)
+
+    try:
+        with open(filename, mode="r", encoding="utf-8") as f:
+            return f.read().rstrip()
+    except IOError as e:
+        # some files with secret must exist, so for such cases it is time
+        # to inform about improper configuration
+        if raise_on_error:
+            raise
+        # no logging configured yet
+        print("Problem reading secret from file {filename}:", e)
```

Comment:
If the user has a typo on the name of the secret filed `apitoken` we will get this error. Can we also the write the log saying `Verify the provider secret contains` `apitoken`

Reply:
added there @xrajesh